libraries/AP_HAL_Linux/RCInput_Multi.cpp

```diff
@@ -0,0 +1,68 @@
+/// -*- tab-width: 4; Mode: C++; c-basic-offset: 4; indent-tabs-mode: nil -*-
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+/*
+  this is a driver for multiple RCInput methods on one board
+ */
+
+#include <AP_HAL/AP_HAL.h>
+
+#if CONFIG_HAL_BOARD_SUBTYPE == HAL_BOARD_SUBTYPE_LINUX_DISCO
+#include "RCInput_Multi.h"
+
+extern const AP_HAL::HAL& hal;
+
+using namespace Linux;
+
+// constructor
+RCInput_Multi::RCInput_Multi(uint8_t _num_inputs, ...) :
```

Comment:
minor, but I don't really like this use of vararg. There's no check _num_inputs actually match the number of arguments. Instead of doing this can we simply pass an std::vector?

See here as an example: https://github.com/ArduPilot/ardupilot/blob/master/libraries/AP_HAL_Linux/I2CDevice.cpp#L274

And how to call it:
https://github.com/ArduPilot/ardupilot/blob/master/libraries/AP_HAL_Linux/HAL_Linux_Class.cpp#L189